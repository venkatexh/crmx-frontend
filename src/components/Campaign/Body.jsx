import React from "react";
import "../../sass/components/contact/contact.component.scss";

const Body = ({ campaign }) => {
  const { subject, text, tags } = campaign;
  
  return (
    <div className={"contactBody"}>
      <div className={"left"}>
        <div className={"heading"}>Email Content</div>
        <div>Subject</div>
        <div>{subject}</div>
        <div>Email Text</div>
        <div>{text}</div>
      </div>
      <div className={"right"}>
        <div className={"heading"}>Tags</div>
        <div className={"tagsList"}>
          {tags?.map((tag, idx) => {
            return (
              <div className={"contactTag"} key={idx}>
                {tag.name}
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default React.memo(Body);
