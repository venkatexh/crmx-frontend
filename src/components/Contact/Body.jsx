import "../../sass/components/contact/contact.component.scss";
import hostHeader from "../../config/host";
import Axios from "axios";
import { updateContact } from "../../redux/actions/contact/updateContact";
import { useDispatch } from "react-redux";
import Activity from "./Activity";

const Body = ({ contact }) => {
  const { tags, activities } = contact;
  const dispatch = useDispatch();

  const handleTagRemoval = (tagId) => {
    Axios.delete(`${hostHeader.url}/api/contact`, {
      data: {
        contactId: contact._id,
        tagId,
      },
    })
      .then((r) => {
        if (r.status === 200) {
          dispatch(updateContact(r.data));
        }
      })
      .catch((err) => {
        console.log(err);
      });
  };

  return (
    <div className={"contactBody"}>
      <div className={"left"}>
        <div className={"heading"}>Activity</div>
        <div className={"activityContainer"}>
          {activities?.map((activity) => (
            <Activity activity={activity} key={activity.id} />
          ))}
        </div>
      </div>
      <div className={"right"}>
        <div className={"heading"}>Tags</div>
        <div className={"tagsList"}>
          {tags?.map((tag, idx) => {
            return (
              <div className={"contactTag"} key={idx}>
                <div className={"tagName"}>{tag.name}</div>
                <div
                  className={"cross"}
                  onClick={() => handleTagRemoval(tag._id)}
                >
                  &#10005;
                </div>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default Body;
