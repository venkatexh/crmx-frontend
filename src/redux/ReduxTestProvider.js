import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
