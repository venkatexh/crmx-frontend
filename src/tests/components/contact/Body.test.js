import { shallow } from "enzyme";
import React from "react";
import Body from "../../../components/Contact/Body";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("Contact body component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <Body
          contact={{
            tags: ["test tag"],
            activities: [{}, {}],
            _id: "testid123",
          }}
        />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
