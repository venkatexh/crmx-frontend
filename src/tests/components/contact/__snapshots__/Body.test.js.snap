// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact body component snapshot 1`] = `ShallowWrapper {}`;
